reference/best_practices/reusable_bundles.rst

```diff
@@ -1,4 +1,173 @@
+
+
+Compatibility
+-------------
+
+Your bundle won't be reusable for some customers if you forget some cases. That is no more an issue if you document compatibilities of your bundle.
+
+Akeneo PIM storage stack comes with MySQL and hybrid MySQL/MongoDB (:doc:`/reference/technical_information/choose_database`).
+You can refer to the internal API to be sure you develop your bundle in the right way.
+Follow these cookbooks for more details: ":doc:`/cookbook/catalog/product/index`" and ":doc:`/cookbook/catalog/common/index`".
+
+There are two existing distributions CE/EE. The main project is `Akeneo PIM Community Edition`_ (CE) but
+we also have an `Enterprise Edition`_ (EE) which comes with several of extra features dedicated to advanced needs.
+
+Many attribute types are managed (14 in CE and one more in EE). Make sure you do not forget any use case for an attribute type.
+
+Try your bundle with other catalogs. Don't forget use cases with many locales, many channels and various catalog volumes.
+You can use our `data generator bundle`_ to generate a catalog with random data.
+
+.. _Akeneo PIM Community Edition: https://github.com/akeneo/pim-community-standard
+.. _Enterprise Edition: https://www.akeneo.com/enterprise-edition/
+.. _data generator bundle: https://github.com/akeneo-labs/DataGeneratorBundle
+
+
+Maintainability
+---------------
+
+Keep in mind the `Single Responsibility`_ from the five `SOLID principles`_ of Object Oriented Programming.
+It seems simple but we often forget to explode your code in multiple classes to make it more readable and testable.
+
+You can use `PMD`_ to help make your code simple, flexible and easily testable.
+
+
+In Akeneo PIM, we use `PhpSpec`_ as unit testing tool and `Behat`_ for behavior-oriented tests.
+
+Use a Continuous Integration (CI) tool to deploy your application, test your code and improve its quality.
+On our open-source projects, we use free CI tools `Travis`_ for unit tests and `Scrutinizer`_ for static analysis.
+You can find some examples on our `Akeneo-Labs`_ projects (`scrutinizer.yml`_ and `travis.yml`_).
+
+.. _Single Responsibility: https://en.wikipedia.org/wiki/Single_responsibility_principle
+.. _SOLID principles: https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)
+.. _PMD: https://phpmd.org/
+.. _PhpSpec: http://phpspec.readthedocs.org/
+.. _Behat: http://docs.behat.org/
+.. _Travis: https://travis-ci.org/
+.. _Scrutinizer: https://scrutinizer-ci.com/
+.. _Akeneo-Labs: https://github.com/akeneo-labs
+.. _scrutinizer.yml: https://github.com/akeneo-labs/CustomEntityBundle/blob/master/.scrutinizer.yml
+.. _travis.yml: https://github.com/akeneo-labs/CustomEntityBundle/blob/master/.travis.yml
+
+
+Another important point about maintainability is to try to rely on interfaces not on concrete classes.
+An interface is a contract you will inject in a method. It will ease you the migration on future Akeneo PIM versions.
```

Comment:
I hesitated a lot and it was in the extensibility part at the begining. I moved it here as I linked it to migration.